lib/web_console/middleware.rb

```diff
@@ -58,6 +59,25 @@ def call(env)
 
     private
 
+      # Returns whether the response is from an acceptable content type.
+      #
+      # We can render a console for HTML and TEXT by default.
+      def acceptable_response_content_type?(headers, env)
```

Comment:
It didn't move it out because I needed the Rack headers, at least it was not the only reason. Since we are checking **response** content type, it does not make sense to include it in the `Request` class. For me, it would lead to confusion between request and response concepts.

Let me try to create a new `Response` class (a wrapper around Rack::Response) adding this specific checking, following the same approach you already did for the requests. I think it would be a nice fit in terms of design.